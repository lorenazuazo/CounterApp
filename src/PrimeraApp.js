import React from 'react';
import PropTypes from 'prop-types';
//<> y cerrado con </>
const PrimeraApp = ({saludo, subtitulo}) => {

    //const saludo = 'Hola Mundo'
    const obj = {
        nom:'lore',
        ap:'zuazo',
    }
    return (
        <>
            <h1>{saludo}!!!!</h1>
            {/* <pre>{JSON.stringify(obj,null,3)}</pre> */}
            {/* <p>Primera App</p> */}
            <p>{subtitulo}</p>
        </>
    )
}
//esto es para exigir a los props
PrimeraApp.propTypes = {
    saludo: PropTypes.string.isRequired
}
//props por default en ved de parametro
PrimeraApp.defaultProps = {
    subtitulo:'es titulo default'
}
export default PrimeraApp;
