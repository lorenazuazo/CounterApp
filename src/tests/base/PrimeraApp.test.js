import React from "react"; //a mi no me lo pidio
import "@testing-library/jest-dom";
import PrimeraApp from "../../PrimeraApp";
import { shallow } from "enzyme";

describe("Pruebas a <PrimeraApp />", () => {

  test('Debe mostar <PrimeraApp /> correctaente', () => {
    const saludo = "Hola, soy Lorena";
    const wrapper = shallow(<PrimeraApp saludo={saludo} />);

    expect(wrapper).toMatchSnapshot();
  });

  test("debe mostrar el subtitulo enviado por props", () => {
    const saludo = "Hola soy Lorena";
    const subtitulo = "Soy un subtituo";

    const wrapper = shallow(
      <PrimeraApp saludo={saludo} subtitulo={subtitulo} />
    );
    //funciona como document.getQuerySelector()
    //se puede buscar por id o clase, y se puede buscar en vez de txt html() por ej
    const textParrafo = wrapper.find("p").text();
   
    expect(textParrafo).toBe(subtitulo);
  });
  
});
