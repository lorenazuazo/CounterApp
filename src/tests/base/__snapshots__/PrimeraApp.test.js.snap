// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas a <PrimeraApp /> Debe mostar <PrimeraApp /> correctaente 1`] = `
<Fragment>
  <h1>
    Hola, soy Lorena
    !!!!
  </h1>
  <p>
    es titulo default
  </p>
</Fragment>
`;

exports[`Pruebas a <PrimeraApp /> Debe mostarel mensaje "Hola, soy Lorena" 1`] = `
<Fragment>
  <h1>
    Hola, soy Lorena
    !!!!
  </h1>
  <p>
    es titulo default
  </p>
</Fragment>
`;

exports[`Pruebas a Primera App Debe mostarel mensaje "Hola, soy Lorena" 1`] = `
<Fragment>
  <h1>
    Hola, soy Lorena
    !!!!
  </h1>
  <p>
    es titulo default
  </p>
</Fragment>
`;
